### DISCUSSION TITLE
custom domain nginx proxy help

### DISCUSSION THREAD

**Discussion by santosh0787** [danny-avila/LibreChat#2395] [Help Wanted]
Team,

I have windows laptop and installed libre chat on docker. When I access librechat url http://localhost:3080, login is working fine, images from Dall-E are loading fine. 

below is my server config:

` #==================================================#
#               Server Configuration               #
#==================================================#

HOST=0.0.0.0
PORT=3080

MONGO_URI=mongodb://127.0.0.1:27017/LibreChat

DOMAIN_CLIENT=http://localhost:3080
DOMAIN_SERVER=http://localhost:3080`

But I have kept nginx proxy server on top of it and below is my config: 

`                    server_name ai.local;
    location /api {
        proxy_pass http://localhost:3080/api; # 
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }

    location / {
        proxy_pass http://localhost:3080; # 
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }  `

- ai.local is my local dns. Pi hole is my dns server. 
- I have added proxy_params based on librechat documentation. 
- I don't have any authentication enabled at nginx side. 
- This is for home purpose so i don't intend to expose to internet. so i'm using http only. 

In this scenario, I'm able to login to http://ai.local but when I prompt for images, they won't load. Every time I refresh the page it is redirecting to login page. 

When I inspect the image and access the image url for example "http://ai.local/images/661922673661e4218962f221/img-0e780b88-526d-4e4b-b45c-6b332d453c0e.png" , im getting unauthorized error. 

Can someone please guide me how to setup in this scenario. My intention is to access libre chat url with custom domain which I use it only for home purpose.

**Comment by danny-avila** [comment]
You can use a Caddyfile for a local domain. I have no trouble using a real cloudflare domain with NGINX in my setup but you can also try Traefik: https://docs.librechat.ai/deployment/traefik.html
